File: leet_0506_relative_ranks.cc
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score)
    {
        //  way 1 : sorting
        //  (still looking for another way)

        vector<int> sorted = score;
        sort(sorted.begin(), sorted.end(), greater<int>());
        // for (int n: sorted)
        //     cout << n << ' ';
        vector<string>  res;
        for (int n: score)
        {
            int idx = find(sorted.begin(), sorted.end(), n) - sorted.begin();
            switch (idx)
            {
                case 0:
                    res.push_back("Gold Medal");
                    break;
                case 1:
                    res.push_back("Silver Medal");
                    break;
                case 2:
                    res.push_back("Bronze Medal");
                    break;
                default:
                    res.push_back(to_string(idx + 1));
                    break;
            }
        }
        return res;
    }
};

File: leet_0506_relative_ranks.ts
function findRelativeRanks(score: number[]): string[] {
    let sorted:number[] = [ ... score]
    sorted.sort((a, b) => b - a)
    let res: string[] = []
    for (let n of score) {

        const idx: number = sorted.indexOf(n);
        switch (idx) {
            case 0:
                res.push("Gold Medal")
                break
            case 1:
                res.push("Silver Medal")
                break
            case 2:
                res.push("Bronze Medal")
                break
            default:
                res.push((idx + 1).toString())
                break
        }
    }
    return res
};
